### PR TITLE
shields: seeed_w5500: new shield

### PR DESCRIPTION
**Pull request by mniestroj**
Support "W5500 Ethernet Shield" manufactured by Seeed Studio.

[1] https://www.seeedstudio.com/W5500-Ethernet-Shield.html

### CODE REVIEW

File: boards/shields/seeed_w5500/Kconfig.defconfig [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# Copyright (c) 2024 Marcin Niestroj
```
Comment (jilaypandya) [inline boards/shields/seeed_w5500/Kconfig.defconfig:1]: Nice to have: SPDX-FileCopyrightText:

File: boards/shields/seeed_w5500/doc/index.rst [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+###########################
+
+Overview
+********
+
+Seeed `W5500 Ethernet Shield`_ is Arduino connector shield with `W5500`_ 10/100
+MBPS stand alone Ethernet controller with on-board MAC & PHY, 16 KiloBytes for
+FIFO buffer and SPI serial interface. This shield additionally has Grove UART
+connector, Grove I2C connector and SD card slot.
+
```
Comment (jilaypandya) [inline boards/shields/seeed_w5500/doc/index.rst:13]: Seeed `W5500 Ethernet Shield`_ is an Arduino connector shield with:
- `W5500`_ 10/100 MBPS stand alone Ethernet controller with on-board
MAC & PHY and 16 KiloBytes for FIFO buffer
- SPI serial interface
- Grove UART connector
- Grove I2C  connector
- SD card slot.

File: boards/shields/seeed_w5500/Kconfig.defconfig (hunk 1 above)
Comment (mniestroj) [reply boards/shields/seeed_w5500/Kconfig.defconfig:1]: Thanks for suggestion. Haven't found any `SPDX-FileCopyrightText:` use in `boards/` directory yet. I would rather leave it as is for now just for consistency.

File: boards/shields/seeed_w5500/doc/index.rst (hunk 2 above)
Comment (mniestroj) [reply boards/shields/seeed_w5500/doc/index.rst:17]: Applied, thanks!

File: boards/shields/seeed_w5500/doc/seeed_w5500.jpg
Comment (nordicjm) [inline boards/shields/seeed_w5500/doc/seeed_w5500.jpg:1]: Convert to webp then put through https://tinypng.com/

**Review by nordicjm — CHANGES_REQUESTED** [review]

File: boards/shields/seeed_w5500/doc/seeed_w5500.jpg
Comment (mniestroj) [reply boards/shields/seeed_w5500/doc/seeed_w5500.jpg:1]: Done!

**Review by nordicjm — APPROVED** [review]

**Review by kartben — APPROVED** [review]

**Pull request merged**